.travis.yml

```diff
@@ -6,10 +22,56 @@ matrix:
           dist: trusty
           sudo: required
           env: TRAVIS_BUILD_TYPE=Debug
+          env: CMAKE_OPTIONS="-DCMAKE_C_COMPILER=/usr/lib/ccache/$CC -DCMAKE_CXX_COMPILER=/usr/lib/ccache/$CXX"
+          env: ETH_PATH="eth"
+          env: DOCS_COMMAND="cd docs && sphinx-build -nW -b html -d _build/doctrees . _build/html && cd .."
+        - os: osx
+          osx_image: xcode7.1
+          env: TRAVIS_BUILD_TYPE=Debug
+          env: CMAKE_OPTIONS="-DCMAKE_C_COMPILER=/usr/lib/ccache/$CC -DCMAKE_CXX_COMPILER=/usr/lib/ccache/$CXX"
+          env: ETH_PATH="~/ethbin/eth"
+          env: DOCS_COMMAND=""
         - os: osx
           osx_image: xcode7.3
           env: TRAVIS_BUILD_TYPE=Debug
+          env: CMAKE_OPTIONS=""
+          env: ETH_PATH="~/ethbin/eth"
+          env: DOCS_COMMAND=""
 git:
     depth: 2
-before_install: ./install_dependencies.sh
-script: ./fetch_umbrella_build_and_test.sh
\ No newline at end of file
+branches:
+    except:
+        - /solidity-appveyor-v[0-9]/
+before_install:
+    - ./install_deps.sh
+before_script:
+    - cd ..
```

Comment:
AFAIK, if you use this dash-syntax, it will just "resume next" on a failed command which is not really what we want.